### ISSUE TITLE
samtools index should reject invalid input files

### ISSUE THREAD

**Issue by holgerbrandl** [samtools/samtools#465]
This should throw an error:
```
touch foobar
samtoools index foobar
```
Same for other non bam/cram/sam files.

At the moment index silently fails/stops without doing anything

**Comment by SamStudio8** [comment]
An empty file, although useless, is valid. But you could use the new
`samtools quickcheck` command (https://github.com/samtools/samtools/pull/406)
to catch these cases.

**Comment by holgerbrandl** [comment]
So if an empty file is a valid bam file, shouldn't `index` create a corresponding index bai-file then? At the moment it does not create anything.

Imho, `index` should nevertheless reject invalid inputs. Eg.
```
echo foo > bar.bam
samtools index bar.bam
```
which just silently fails without any complaint or error.

**Comment by SamStudio8** [comment]
This was originally reported in #362, it looks like @jmarshall fixed this with 9bbc79e on the dev branch, so a fix will be in the next version of samtools.

> ```
$ echo hello > hello.bam
$ samtools index hello.bam || echo FAILED
[bam_index] corrupted or unsorted input file
FAILED
```

**Comment by jmarshall** [comment]
Thanks for the bug report; as Sam notes, already fixed (post-1.2).
